src/main/java/com/shapesecurity/shift/es2016/codegen/CodeGen.java

```diff
@@ -180,12 +180,11 @@ public CodeRep reduceArrayExpression(@Nonnull ArrayExpression node, @Nonnull Imm
     @Nonnull
     @Override
     public CodeRep reduceArrowExpression(@Nonnull ArrowExpression node, @Nonnull CodeRep params, @Nonnull CodeRep body) {
-        if (isComplexArrowHead(node.params)) {
-            params = factory.paren(params);
+        if (!isComplexArrowHead(node.params)) {
+            // FormalParameters unconditionally include parentheses, but they're not necessary here
+            params = this.reduceBindingIdentifier((BindingIdentifier) node.params.items.maybeHead().fromJust());
         }
-        if (node.body instanceof FunctionBody) {
-            body = factory.brace(body);
-        } else if (body.startsWithCurly()) {
+        if (body.startsWithCurly()) {
```

Comment:
Why is this not `startsWithObjectCurly`?

Reply:
'Cause I forgot. Fixed.